test/extrema_test.dart

```diff
@@ -0,0 +1,217 @@
+    expect(findLastMax.val.value.toInt(), equals(13));
+    expect(findLastMax.index.value.toInt(), equals(5));
+  });
+
+  test('List with same extrema including max in first and last.', () async {
+    // Create a list of Logic objects with different values.
+    final logics = [
+      Logic(width: 8)..put(LogicValue.ofString('000001101')), // 13 in decimal
+      Logic(width: 4)..put(LogicValue.ofString('1101')), // 13 in decimal
+      Logic(width: 8)..put(LogicValue.ofString('00000100')), // 4 in decimal
+      Logic(width: 2)..put(LogicValue.ofString('11')), // 3 in decimal
+      Logic(width: 8)..put(LogicValue.ofString('00001100')), // 12 in decimal
+      Logic(width: 6)..put(LogicValue.ofString('000011')), // 3 in decimal
+      Logic(width: 8)..put(LogicValue.ofString('000001101')), // 13 in decimal
+    ];
+
+    // Create an instance of FindMin.
+    final findFirstMin = Extrema(logics, max: false);
+    await findFirstMin.build();
+
+    // Create an instance of FindMax.
+    final findFirstMax = Extrema(logics);
+    await findFirstMax.build();
+
+    // Create an instance of FindMin.
+    final findLastMin = Extrema(logics, max: false, first: false);
+    await findLastMin.build();
+
+    // Create an instance of FindMax.
+    final findLastMax = Extrema(logics, first: false);
+    await findLastMax.build();
+
+    // Verify the min value and index
+    expect(findFirstMin.val.value.toInt(), equals(3));
+    expect(findFirstMin.index.value.toInt(), equals(3));
+
+    // Verify the max value and index.
+    expect(findFirstMax.val.value.toInt(), equals(13));
+    expect(findFirstMax.index.value.toInt(), equals(0));
+
+    // Verify the min value and index
+    expect(findLastMin.val.value.toInt(), equals(3));
+    expect(findLastMin.index.value.toInt(), equals(5));
+
+    // Verify the max value and index.
+    expect(findLastMax.val.value.toInt(), equals(13));
+    expect(findLastMax.index.value.toInt(), equals(6));
+  });
+
+  test('List containing one element.', () async {
+    // Create a list of Logic objects with different values
+    final logics = [
+      Logic(width: 4)..put(LogicValue.ofString('1100')), // 12 in decimal
+    ];
+
+    // Create an instance of FindMin.
+    final findMin = Extrema(logics, max: false);
+    await findMin.build();
+
+    // Create an instance of FindMax
+    final findMax = Extrema(logics);
+    await findMax.build();
+
+    // Verify the minimum value and index
+    expect(findMax.val.value.toInt(), equals(12));
+    expect(findMax.index.value.toInt(), equals(0));
+  });
+
+  test('List containing an empty element.', () async {
+    // Create a list of Logic objects with different values
+    final logics = [Logic(width: 4)]; // Empty element
+
+    // Create an instance of FindMax
+    final findMax = Extrema(logics);
+    await findMax.build();
+
+    // Verify the minimum value and index
+    expect(() => findMax.val.value.toInt(), throwsA(isA<RohdException>()));
```

Comment:
this test is not interesting because the exception is related to converting an illegal value (Z or X) into an integer, not something related to your component.  It's implicit that invalid inputs to your module may produce invalid outputs.